pkg/steps/template.go

```diff
@@ -728,6 +731,15 @@ func podJobIsFailed(pod *coreapi.Pod) bool {
 	return false
 }
 
+func isPodRunning(pod *coreapi.Pod) bool {
+	for _, condition := range pod.Status.Conditions {
```

Comment:
IMHO, there can be a time where the pod was ready but failed after. Searching through pod's condition history to find if there is a `ready` status, shouldn't be the case here. Instead, I think it is better to check only the most recent's time condition.

Reply:
There can only be one ready condition